src/blocks/transformers/modalForm/modal.tsx

```diff
@@ -0,0 +1,130 @@
+  }
+
+  inputSchema: Schema = {
+    type: "object",
+    properties: {
+      schema: {
+        type: "object",
+        description: "The JSON Schema for the form",
+        additionalProperties: true,
+      },
+      uiSchema: {
+        type: "object",
+        description: "The react-jsonschema-form uiSchema for the form",
+        additionalProperties: true,
+      },
+      cancelable: {
+        type: "boolean",
+        description:
+          "Whether or not the user can cancel the form (default=true)",
+        default: true,
+      },
+      submitCaption: {
+        type: "string",
+        description: "The submit button caption (default='Submit')",
+        default: "Submit",
+      },
+    },
+    required: ["schema"],
+  };
+
+  async transform({
+    schema,
+    uiSchema = {},
+    cancelable = true,
+    submitCaption = "Submit",
+  }: BlockArg): Promise<unknown> {
+    // Future improvements:
+    // - Bootstrap 4 modals prevent page scroll when the modal is showing
+    // - Support draggable modals. This will require showing the modal header on the host page so there's a drag handle?
+
+    expectContext("contentScript");
+
+    const container = document.createElement("div");
+    const shadowRoot = container.attachShadow({ mode: "closed" });
+
+    const nonce = uuidv4();
+    const id = `modal-${nonce}`;
+
+    const tab = await whoAmI();
+
+    const frameSrc = new URL(browser.runtime.getURL("modalForm.html"));
+    frameSrc.searchParams.set("nonce", nonce);
+    frameSrc.searchParams.set("frameId", String(tab.frameId));
+
+    // By setting the modal-content 100vh, the iframe form content can expand to fit the available vertical size as
+    // needed. Otherwise, we'd need to have the form message the content script with a requested vertical height.
+    // (We might do this anyway in the future to support draggable modals)
+    $(shadowRoot).append(`
+        <link rel="stylesheet" href="${theme}" />
+        <div class="modal-backdrop show"></div>
+        <div id="${id}" class="modal show" style="display: block">
+            <div class="modal-dialog modal-dialog-scrollable" role="document">
+                <div class="modal-content" style="height: 100vh; background: none transparent; border: none;">
+                    <iframe src="${frameSrc.toString()}" frameborder="0" allowtransparency="true" width="100%" height="100%"/>
+                </div>
+            </div>
+        </div>
+    `);
+
+    document.body.append(container);
+
+    const modal = shadowRoot.querySelector("#" + id);
+
+    if (modal) {
+      // Try to intercept click handler on the host page. Doesn't work on all sites, e.g., Trello that define a
+      // `click` handler on document with `useCapture: true`.
+      // See https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener
+
+      modal.addEventListener("click", (e) => {
+        e.stopPropagation();
+        e.stopImmediatePropagation();
+      });
+    }
```

Comment:
I'd just use a `!`, the modal should always exist since the code just created it

```suggestion
    // Try to intercept click handler on the host page. Doesn't work on all sites, e.g., Trello that define a
    // `click` handler on document with `useCapture: true`.
    // See https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener

    const modal = shadowRoot.querySelector("#" + id)!;
    modal.addEventListener("click", (e) => {
      e.stopPropagation();
      e.stopImmediatePropagation();
    });
```